src/firefly/js/ui/FieldGroup.jsx

```diff
@@ -63,9 +65,11 @@ export class FieldGroup extends Component {
         if (shallowequal(this.props, nextProps)) return false;
         const wrapperGroupKey= nextContext;
         const {groupKey, reducerFunc, keepState, actionTypes}= nextProps;
-        if (this.props.groupKey!==groupKey) {// support change the groupKey property on the form without unmounting
+        // added check for 'reducerFunc.ver' to re-register if needed.  Should revisit this for a better solution
+        if (this.props.groupKey!==groupKey || this.props?.reducerFunc?.ver !== reducerFunc?.ver) {// support change the groupKey property on the form without unmounting
```

Comment:
After studying all this, I think it would be better to add a `dispatchUpdateReducer`.
```js
if (this.props.groupKey!==groupKey) 
   dispatchMountFieldGroup(groupKey, false);
   dispatchMountFieldGroup(groupKey, true, keepState, null, reducerFunc, actionTypes, wrapperGroupKey);
}
else if (this.props?.reducerFunc !== reducerFunc) {
   dispatchUpdateReducer(groupKey,reducerFunc)
}
```
The reducer would be something like:
```js
const fg= {...state[groupKey], reducerFunc};
return {....state, [groupKey]: {...fg, fields: fireFieldsReducer(fg, action)}
```
I will do this on Monday and push to this branch, then you can try it out. If it does not work we can revert. It would make the external code simpler.